python/lsst/eo/pipe/biasShiftsTask.py

```diff
@@ -0,0 +1,286 @@
+        shift_peaks.append(region[0]+np.argmax(motion[region[0]:region[1]]))
+    if len(shift_regions)==0: return np.array([np.NaN]*4)
+    return np.concatenate((np.array([shift_peaks]),shift_regions.T)).T
+
+def flat_kernel(window=30):
+    kernel = np.concatenate([np.ones(window), -1*np.ones(window-1)])
+    return kernel/window
+
+def odd_local_kernel(window=30):
+    kernel = np.concatenate([np.arange(window), np.arange(-window+1,0)])
+    kernel = kernel/np.sum(kernel[:window])
+    return kernel
+
+def shift_kernel(window=30):
+    return odd_local_kernel(window)
+
+def scan_for_shifts(data, window=30, noise_filter=30, threshold=3, skip_rows=30):
+    """
+    Looks for shifts in baseline in noisy data using a convolution.
+    
+    Returns
+    -------
+    An array of detected shifts with rows [peak of shift, start of shift area, end of shift area]
+    
+    """
+    local_noise = np.std(butter_highpass_filter(data, 1/noise_filter,1))
+    shift_conv = np.convolve(data, shift_kernel(window),mode='valid')
+    shift_conv = np.concatenate([np.zeros(window-1), shift_conv, np.zeros(window)])
+    shift_like = np.abs(shift_conv)/local_noise
+    shift_mask = shift_like > threshold
+    shift_mask[:skip_rows]=False
+    shift_regions = find_mask_runs(shift_mask)
+    shift_peaks = []
+    for region in shift_regions:
+        region_peak = np.argmax(shift_like[region[0]:region[1]]) + region[0]
+        if satisfies_flatness(region_peak, shift_conv[region_peak], data):
+            shift_peaks.append([shift_conv[region_peak],region_peak, region[0], region[1]])
+    if len(shift_peaks)==0: return np.array([[np.NaN]*4]), local_noise
+    return np.asarray(shift_peaks),  local_noise
+
+def satisfies_flatness(shiftrow, shiftmag, oscans, window=30, verbose=False):
+    
+    prerange = np.arange(shiftrow-window, shiftrow)
+    postrange = np.arange(shiftrow, shiftrow+window)
+    
+    preslope, preintercept, pre_r_value, p_value, std_err = scipy.stats.linregress(prerange,oscans[prerange])
+    postslope, postintercept, post_r_value, p_value, std_err = scipy.stats.linregress(postrange,oscans[postrange])
+    
+    if verbose:
+        print(f'Pre-threshold: {preslope*2*len(prerange)}')
+        print(f'Post-threshold: {postslope*2*len(postrange)}')
+        print(f'Shift value: {shiftmag}')
+    
+    pretrending  = (preslope*2*len(prerange) < shiftmag)   if (shiftmag>0) else (preslope*2*len(prerange) > shiftmag)
+    posttrending = (postslope*2*len(postrange) < shiftmag) if (shiftmag>0) else (postslope*2*len(postrange) > shiftmag)
+    
+    return (pretrending and posttrending)
+
+def find_shifts_in_exposure(exp, amp=None, method="conv", **options):
+    methods = ["derivative","conv"]
+    if method not in methods:
+        raise Exception(f'Shift detection method "{method}" not understood. Available methods are {methods}')        
+    if amp is None:
+        det = exp.getDetector()
+    else: 
+        det = [amp.getName()]
```

Comment:
This looks wrong.  Here `det` is a list of a single string, but above `det` is an actual detector object.  It seems like this should be 
```
det = [amp]
```